pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/ImageSrcTransformer.java

```diff
@@ -0,0 +1,52 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.asciidoctor.ast.Document;
+import org.asciidoctor.extension.Postprocessor;
+import org.jsoup.Jsoup;
+
+import java.util.Base64;
+
+import static com.redhat.pantheon.conf.GlobalConfig.IMAGE_PATH_PREFIX;
+
+/**
+ * An asciidoctor post processor which transforms all image locations in the resulting html to an encoded and absolutely
+ * resolving path comprised of a prefix and a unique identifier for the image. This unique identifier consists of the
+ * Base64 encoding of the actual absolute path of the image resource in the JCR repository.
+ *
+ * @see com.redhat.pantheon.servlet.assets.ImageServletFilter for the servlet serving these paths
+ * @author Carlos Munoz
+ */
+public class ImageSrcTransformer extends Postprocessor {
+
+    private final Resource module;
+
+    public ImageSrcTransformer(Resource module) {
+        this.module = module;
+    }
+
+    @Override
+    public String process(Document document, String output) {
+        org.jsoup.nodes.Document doc = Jsoup.parse(output, "UTF-8");
+
+        ResourceResolver resourceResolver = module.getResourceResolver();
+
+        // add a prefix to all images
+        doc.select("img")
+                .forEach(imageElement -> {
+                    String imgSrc = imageElement.attr("src");
+                    Resource resolvedImage = resourceResolver.getResource(module.getParent(), imgSrc);
+                    if(resolvedImage != null) {
```

Comment:
Are you sure we want to be making this check? I just tested a file with a broken image link (pointing at ./images/tiger.png instead of ./tiger.png where the file was actually located) and was confused for some time why the code appeared not to be doing the substitution. I don't think we gain anything by skipping the logic in this scenario. It's broken either way - let's at least make the behavior uniform, unless you can think of a string argument why we should not do so.

Reply:
makes sense... I'll remove the check.

Reply:
Ok, I spoke to soon on this one too.

I still need to make the check, because I need to encode the absolute path of the resource, and for that I need the resource not to be null.
What I could do in case of the resource not being found is to encode whatever path is in the image. This would ensure all images are in the same format, but it would add an extra step to debugging a non-existing image, because a developer would have to decode the path to get to the actual path being resolved. Not a big deal, so I'm still good if you want to encode all paths.

Reply:
Hmm. Let me think on this for a bit.